test/benchmark/benchmark.go

```diff
@@ -164,24 +200,60 @@ func tearDownWeavaite() error {
 // start weaviate in case it was not already started
 //
 // We want to benchmark the current state and therefore need to rebuild and then start a docker container
-func startWeavaite(c *http.Client, url string) (bool, error) {
-	requestReady, _ := http.NewRequest("GET", url+".well-known/ready", bytes.NewReader([]byte{}))
-	requestReady.Header.Set("content-type", "application/json")
-	response_started, err := c.Do(requestReady)
-	if err == nil {
-		response_started.Body.Close()
-	}
-	alreadyRunning := err == nil && response_started.StatusCode == 200
+func startWeaviate(c *http.Client, url string) bool {
+	requestReady := createRequest(url+".well-known/ready", "GET", nil)
+
+	response_started_code, _, _, err := performRequest(c, requestReady)
```

Comment:
I think that here you forgot about error handling, what about a case: `if err != nil`?

Reply:
Instead of checking for an error, it checks for "successfully responding" and returns in that case. If `err != nil` or the statuscode was not 200 the tools/test/run_ci_server.sh script is executed which kills all running weaviate instances and starts a new one.